src/Environment.cpp

```diff
@@ -228,6 +229,16 @@ namespace geopm
         return lookup("GEOPM_AGENT");
     }
 
+    double EnvironmentImp::period(double default_period) const
+    {
+        double result = default_period;
+        std::string period_str = lookup("GEOPM_PERIOD");
```

Comment:
Do exception management for stod() throwing on invalid input.  Throw and give a useful message.